src/overlays/actors/ovl_En_Ani/z_en_ani.c

```diff
@@ -28,62 +44,314 @@ const ActorInit En_Ani_InitVars = {
+        Quake_SetSpeed(quakeValue, 0x6978);
+        Quake_SetQuakeValues(quakeValue, 7, 0, 0, 0);
+        Quake_SetCountdown(quakeValue, 0x14);
+        Audio_PlayActorSound2(&this->actor, NA_SE_IT_HAMMER_HIT);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/func_809679D0.s")
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 2, 2000, 0x100);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/func_80967A48.s")
+void EnAni_LosingBalance(EnAni* this, GlobalContext* globalCtx) {
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/func_80967AB4.s")
+    if (SkelAnime_Update(&this->skelAnime)) {
+        this->actor.minVelocityY = -20.0f;
+        this->actor.gravity = -5.0f;
+        this->actor.velocity.y = 0.0f;
+        this->actor.velocity.z = -4.0f;
+        // frame count : 0.0f, only first frame, rest is handled in next action func
+        Animation_Change(&this->skelAnime, &gAniLandingThenStandingUpAnim, 0.0f, 0.0f, 0.0f, 2, 5.0f);
+        this->actionFunc = EnAni_FallingToGround;
+        gSaveContext.eventInf[1] |= 0x10;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/EnAni_Init.s")
+void EnAni_TreeHanging(EnAni* this, GlobalContext* globalCtx) {
+    if (SkelAnime_Update(&this->skelAnime) != 0) {
+        if (this->treeReachTimer > 0) {
+            this->treeReachTimer--;
+            if (this->treeReachTimer > 0) {
+                Animation_PlayOnce(&this->skelAnime, &gAniTreeHangingReachAnim);
+            } else {
+                Animation_PlayOnce(&this->skelAnime, &gAniTreeHangingAnim);
+            }
+        } else if (Rand_ZeroFloat(1.0f) < 0.4f) {
+            Animation_PlayOnce(&this->skelAnime, &gAniTreeHangingReachAnim);
+            this->treeReachTimer = 2;
+        } else {
+            Animation_PlayOnce(&this->skelAnime, &gAniTreeHangingAnim);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/EnAni_Destroy.s")
+    // his bonk detection is based on his tilt, neat
+    if (this->actor.home.rot.x != 0) {
+        this->stateFlags |= ANI_STATE_FALLING;
+        this->actionFunc = EnAni_LosingBalance;
+        Animation_Change(&this->skelAnime, &gAniTreeHangLosingBalanceAnim, 1.0f, 0.0f,
+                         Animation_GetLastFrame(&gAniTreeHangLosingBalanceAnim), 2, -5.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/func_80967D20.s")
+void EnAni_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnAni* this = THIS;
+    f32 minVelocity;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ani/func_80967DA0.s")
+    Collider_UpdateCylinder(&this->actor, &this->collider1);
+    Collider_UpdateCylinder(&this->actor, &this->collider2);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider1.base);
+    if ((this->stateFlags & ANI_STATE_UNUSED) == 0) {
```

Comment:
```suggestion
    if (!(this->stateFlags & ANI_STATE_UNUSED)) {
```
No need for `== 0` on flag accesses.